scipy/stats/_fit.py

```diff
@@ -826,6 +831,21 @@ def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
     allows the user to perform the Monte Carlo trials adapted to their
     particular data.
 
+    *Algorithmic overview*
+
+    In brief, this routine executes the following steps:
+    
+      1. Fit unknown parameters to the given `data`, thereby forming the
+         "null-hypothesized" distribution and compute the statistic for them.
```

Comment:
"... compute the statistic for this pair (data and distribution)."?

Reply:
I'm not sure. Personally, I'd try to avoid parentheses in documentation because they are often a sign that the rest of the sentence is not phrased as clearly as it could be. Otherwise, why would the parentheses be needed? That said, I'm not always successful, so maybe it's OK here!

On the other hand, you could probably just remove this part. I think the steps would still be very understandable.

```suggestion
         "null-hypothesized" distribution.
```

In step 4, "Calculate the statistic between each sample and the distribution that has been fit to the sample." I think the reader will assume that "each sample" includes `data`. Once they get to step 5, there should be not doubt about whether that was the intended interpretation.

But I leave the decision to you, and I'll merge it either way.

Reply:
I decided for 
```
1. Fit unknown parameters to the given `data`, thereby forming the
   "null-hypothesized" distribution, and compute the statistic between
   this pair of data and distribution.
```